module,execution_time (sec),success,timestamp,error,error_short
cowidev.vax.batch.argentina,19.75,True,2023-07-13T07:00:26,,
cowidev.vax.batch.australia,,,2023-07-13T07:00:06,,
cowidev.vax.batch.austria_old,,,2023-07-13T07:00:06,,
cowidev.vax.batch.belgium,9.5,True,2023-07-13T07:00:16,,
cowidev.vax.batch.bolivia,4.75,True,2023-07-13T07:00:11,,
cowidev.vax.batch.canada,5.87,False,2023-07-13T07:01:55,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 282, in main
    Canada().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 270, in export
    df = df.pipe(self.pipeline, df_man)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 250, in pipeline
    .pipe(self.pipe_make_monotonic)[
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 230, in pipe_make_monotonic
    df = df.pipe(self.make_monotonic, max_removed_rows=self.max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/utils/base.py"", line 120, in make_monotonic
    return mkm(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 23, in make_monotonic
    return make_monotonic_new(df, column_date, column_metrics, max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 96, in make_monotonic_new
    raise Exception(
Exception: 2 chunks of lengths 36, 29 would be NaNed for metric people_vaccinated. That is more than maximum allowed (22) by make_monotonic() - check the data. Check dates ['2023-01-26', '2023-06-15']. Location(s) affected: {'Canada'}
","2 chunks of lengths 36, 29 would be NaNed for metric people_vaccinated. That is more than maximum allowed (22) by make_monotonic() - check the data. Check dates ['2023-01-26', '2023-06-15']. Location(s) affected: {'Canada'}"
cowidev.vax.batch.chile,0.84,True,2023-07-13T07:00:07,,
cowidev.vax.batch.czechia,34.18,True,2023-07-13T07:00:40,,
cowidev.vax.batch.denmark,,,2023-07-13T07:00:06,,
cowidev.vax.batch.ecdc,88.28,True,2023-07-13T07:01:35,,
cowidev.vax.batch.ecuador,,,2023-07-13T07:00:06,,
cowidev.vax.batch.estonia,0.76,True,2023-07-13T07:00:07,,
cowidev.vax.batch.france,8.45,True,2023-07-13T07:00:15,,
cowidev.vax.batch.germany,0.14,True,2023-07-13T07:00:06,,
cowidev.vax.batch.greece,21.12,True,2023-07-13T07:00:27,,
cowidev.vax.batch.hong_kong,4.12,False,2023-07-13T07:01:59,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/hong_kong.py"", line 226, in main
    HongKong().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/hong_kong.py"", line 210, in export
    df = df_base.pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/hong_kong.py"", line 117, in pipeline
    .pipe(self.make_monotonic)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/utils/base.py"", line 120, in make_monotonic
    return mkm(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 23, in make_monotonic
    return make_monotonic_new(df, column_date, column_metrics, max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 96, in make_monotonic_new
    raise Exception(
Exception: 1 chunks of lengths 516 would be NaNed for metric total_vaccinations. That is more than maximum allowed (10) by make_monotonic() - check the data. Check dates ['2023-06-24']. Location(s) affected: {'Hong Kong'}
",1 chunks of lengths 516 would be NaNed for metric total_vaccinations. That is more than maximum allowed (10) by make_monotonic() - check the data. Check dates ['2023-06-24']. Location(s) affected: {'Hong Kong'}
cowidev.vax.batch.indonesia,,,2023-07-13T07:00:06,,
cowidev.vax.batch.ireland,3.89,True,2023-07-13T07:00:10,,
cowidev.vax.batch.israel,0.1,False,2023-07-13T07:01:59,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/israel.py"", line 189, in main
    Israel().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/israel.py"", line 175, in export
    df = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/israel.py"", line 21, in read
    data = request_json(self.source_url, verify=False)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/web/scraping.py"", line 107, in request_json
    return json.loads(text)
  File ""/usr/lib/python3.10/json/__init__.py"", line 346, in loads
    return _default_decoder.decode(s)
  File ""/usr/lib/python3.10/json/decoder.py"", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File ""/usr/lib/python3.10/json/decoder.py"", line 355, in raw_decode
    raise JSONDecodeError(""Expecting value"", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 3 column 2 (char 3)
",Expecting value: line 3 column 2 (char 3)
cowidev.vax.batch.italy,11.45,True,2023-07-13T07:00:18,,
cowidev.vax.batch.japan,9.25,False,2023-07-13T07:02:08,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 375, in main
    Japan().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 359, in export
    df = self.read().pipe(self.pipeline_base)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 71, in read
    df_latest = self.read_latest().pipe(self.pipe_read_latest)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 100, in read_latest
    self._read_latest_booster(),
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 211, in _read_latest_booster
    df = self._read_xlsx(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 233, in _read_xlsx
    raise ValueError(f""Unknown sheets: {sheets_unknown}"")
ValueError: Unknown sheets: {'5.8以降接種_一般接種（６回目）', '5.8以降接種_一般接種（４回目）', '5.8以降接種_一般接種（３回目）', '６回目_総接種回数', '5.8以降接種_一般接種（５回目）', '５回目_一般接種回数', '5.8以降接種_総接種回数', '５回目_総接種回数', '６回目_一般接種回数'}
","Unknown sheets: {'5.8以降接種_一般接種（６回目）', '5.8以降接種_一般接種（４回目）', '5.8以降接種_一般接種（３回目）', '６回目_総接種回数', '5.8以降接種_一般接種（５回目）', '５回目_一般接種回数', '5.8以降接種_総接種回数', '５回目_総接種回数', '６回目_一般接種回数'}"
cowidev.vax.batch.jersey,5.57,True,2023-07-13T07:00:12,,
cowidev.vax.batch.latvia,66.2,True,2023-07-13T07:01:12,,
cowidev.vax.batch.lithuania,5.74,True,2023-07-13T07:00:12,,
cowidev.vax.batch.luxembourg,2.17,True,2023-07-13T07:00:08,,
cowidev.vax.batch.malaysia,0.8,True,2023-07-13T07:00:07,,
cowidev.vax.batch.malta,0.47,False,2023-07-13T07:02:09,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/tools/datetimes.py"", line 509, in _to_datetime_with_format
    values, tz = conversion.datetime_to_datetime64(arg)
  File ""pandas/_libs/tslibs/conversion.pyx"", line 359, in pandas._libs.tslibs.conversion.datetime_to_datetime64
TypeError: Unrecognized value type: <class 'str'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/malta.py"", line 111, in main
    Malta().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/malta.py"", line 106, in export
    df = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/malta.py"", line 98, in pipeline
    .pipe(self.pipe_date)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/malta.py"", line 64, in pipe_date
    return df.assign(date=clean_date_series(df.date, ""%d/%m/%Y""))
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/clean/dates.py"", line 203, in clean_date_series
    ds_new = pd.to_datetime(ds, format=format_input, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/tools/datetimes.py"", line 887, in to_datetime
    values = convert_listlike(arg._values, format)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/tools/datetimes.py"", line 393, in _convert_listlike_datetimes
    res = _to_datetime_with_format(
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/tools/datetimes.py"", line 513, in _to_datetime_with_format
    raise err
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/tools/datetimes.py"", line 500, in _to_datetime_with_format
    res = _array_strptime_with_fallback(
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/tools/datetimes.py"", line 436, in _array_strptime_with_fallback
    result, timezones = array_strptime(arg, fmt, exact=exact, errors=errors)
  File ""pandas/_libs/tslibs/strptime.pyx"", line 150, in pandas._libs.tslibs.strptime.array_strptime
ValueError: time data '1380949' does not match format '%d/%m/%Y' (match)
",time data '1380949' does not match format '%d/%m/%Y' (match)
cowidev.vax.batch.netherlands,,,2023-07-13T07:00:06,,
cowidev.vax.batch.new_zealand,20.97,True,2023-07-13T07:00:27,,
cowidev.vax.batch.norway,0.37,True,2023-07-13T07:00:07,,
cowidev.vax.batch.peru,0.59,True,2023-07-13T07:02:09,,
cowidev.vax.batch.portugal,,,2023-07-13T07:00:06,,
cowidev.vax.batch.romania,39.79,True,2023-07-13T07:00:46,,
cowidev.vax.batch.saudi_arabia,11.0,True,2023-07-13T07:00:17,,
cowidev.vax.batch.singapore,4.58,True,2023-07-13T07:00:11,,
cowidev.vax.batch.slovakia,51.97,True,2023-07-13T07:00:58,,
cowidev.vax.batch.slovenia,2.58,True,2023-07-13T07:00:09,,
cowidev.vax.batch.south_korea,25.23,False,2023-07-13T07:02:34,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 209, in main
    SouthKorea().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 191, in export
    df_base = self.read().pipe(self.pipeline_base)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 109, in pipeline_base
    .pipe(self.pipe_check_metrics)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 92, in pipe_check_metrics
    raise ValueError(f""Metric {dose} for 'all' is not equal to the sum over all vaccines (2% tolerance)."")
ValueError: Metric dose_1 for 'all' is not equal to the sum over all vaccines (2% tolerance).
",Metric dose_1 for 'all' is not equal to the sum over all vaccines (2% tolerance).
cowidev.vax.batch.spc,35.68,True,2023-07-13T07:00:42,,
cowidev.vax.batch.sweden,22.72,True,2023-07-13T07:00:29,,
cowidev.vax.batch.switzerland,81.68,True,2023-07-13T07:01:28,,
cowidev.vax.batch.trinidad_and_tobago,4.98,True,2023-07-13T07:00:12,,
cowidev.vax.batch.ukraine,,,2023-07-13T07:00:07,,
cowidev.vax.batch.united_kingdom,24.0,True,2023-07-13T07:00:31,,
cowidev.vax.batch.united_states,20.72,True,2023-07-13T07:00:28,,
cowidev.vax.batch.uruguay,4.26,True,2023-07-13T07:00:12,,
cowidev.vax.batch.zimbabwe,1.43,True,2023-07-13T07:00:10,,
cowidev.vax.incremental.africacdc,,,2023-07-13T07:00:09,,
cowidev.vax.incremental.antigua_barbuda,,,2023-07-13T07:00:09,,
cowidev.vax.incremental.aruba,1.82,True,2023-07-13T07:00:11,,
cowidev.vax.incremental.austria,,,2023-07-13T07:00:10,,
cowidev.vax.incremental.azerbaijan,,,2023-07-13T07:00:10,,
cowidev.vax.incremental.bahrain,,,2023-07-13T07:00:10,,
cowidev.vax.incremental.bangladesh,98.79,True,2023-07-13T07:01:49,,
cowidev.vax.incremental.barbados,,,2023-07-13T07:00:10,,
cowidev.vax.incremental.brazil,1.36,True,2023-07-13T07:00:12,,
cowidev.vax.incremental.bulgaria,9.68,True,2023-07-13T07:00:20,,
cowidev.vax.incremental.china,19.93,True,2023-07-13T07:00:31,,
cowidev.vax.incremental.costa_rica,,,2023-07-13T07:00:11,,
cowidev.vax.incremental.croatia,1.42,True,2023-07-13T07:00:12,,
cowidev.vax.incremental.cuba,1.3,False,2023-07-13T07:02:36,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/cuba.py"", line 82, in main
    Cuba().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/cuba.py"", line 68, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/cuba.py"", line 29, in read
    return self._parse_data(soup)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/cuba.py"", line 34, in _parse_data
    ""date"": self._parse_date(soup),
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/cuba.py"", line 41, in _parse_date
    date_str = match.group(1)
AttributeError: 'NoneType' object has no attribute 'group'
",'NoneType' object has no attribute 'group'
cowidev.vax.incremental.curacao,,,2023-07-13T07:00:12,,
cowidev.vax.incremental.cyprus,3.78,True,2023-07-13T07:00:16,,
cowidev.vax.incremental.dominican_republic,13.07,False,2023-07-13T07:02:49,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 112, in main
    DominicanRepublic().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 68, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 26, in read
    metrics = self._parse_metrics(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 35, in _parse_metrics
    metric_candidates = _find_potential_metrics(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 91, in _find_potential_metrics
    h3 = _find_h3(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 81, in _find_h3
    raise ValueError(""More than one element found!"")
ValueError: More than one element found!
",More than one element found!
cowidev.vax.incremental.el_salvador,12.04,True,2023-07-13T07:00:24,,
cowidev.vax.incremental.faeroe_islands,1.86,True,2023-07-13T07:00:14,,
cowidev.vax.incremental.fiji,0.0,True,2023-07-13T07:00:12,,
cowidev.vax.incremental.finland,0.9,True,2023-07-13T07:00:13,,
cowidev.vax.incremental.gabon,,,2023-07-13T07:00:12,,
cowidev.vax.incremental.georgia,9.28,True,2023-07-13T07:00:22,,
cowidev.vax.incremental.greenland,,,2023-07-13T07:00:13,,
cowidev.vax.incremental.guatemala,,,2023-07-13T07:00:13,,
cowidev.vax.incremental.guernsey,,,2023-07-13T07:00:13,,
cowidev.vax.incremental.hungary,77.9,True,2023-07-13T07:01:31,,
cowidev.vax.incremental.iceland,3.63,True,2023-07-13T07:00:18,,
cowidev.vax.incremental.india,2.39,True,2023-07-13T07:00:17,,
cowidev.vax.incremental.iran,,,2023-07-13T07:00:16,,
cowidev.vax.incremental.isle_of_man,5.01,True,2023-07-13T07:00:21,,
cowidev.vax.incremental.jamaica,8.84,True,2023-07-13T07:00:25,,
cowidev.vax.incremental.kazakhstan,11.39,True,2023-07-13T07:00:29,,
cowidev.vax.incremental.kosovo,,,2023-07-13T07:00:17,,
cowidev.vax.incremental.kyrgyzstan,4.96,True,2023-07-13T07:00:22,,
cowidev.vax.incremental.laos,1.7,True,2023-07-13T07:02:50,,
cowidev.vax.incremental.macao,,,2023-07-13T07:00:18,,
cowidev.vax.incremental.moldova,,,2023-07-13T07:00:18,,
cowidev.vax.incremental.monaco,79.86,True,2023-07-13T07:01:38,,
cowidev.vax.incremental.mongolia,2.08,True,2023-07-13T07:00:22,,
cowidev.vax.incremental.montenegro,1.56,True,2023-07-13T07:00:22,,
cowidev.vax.incremental.morocco,,,2023-07-13T07:00:21,,
cowidev.vax.incremental.myanmar,,,2023-07-13T07:00:21,,
cowidev.vax.incremental.nepal,,,2023-07-13T07:00:21,,
cowidev.vax.incremental.north_macedonia,,,2023-07-13T07:00:21,,
cowidev.vax.incremental.northern_cyprus,4.67,True,2023-07-13T07:00:25,,
cowidev.vax.incremental.paho,83.19,False,2023-07-13T07:04:14,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 195, in main
    PAHO().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 190, in export
    df = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 167, in pipeline
    df.pipe(self.pipe_check_columns)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 122, in pipe_check_columns
    raise ValueError(f""Missing column fields: {columns_missing}. Present columns are: {df.columns}"")
ValueError: Missing column fields: {'1st additional dose', 'Second dose', 'Total doses', 'Country code', 'Single dose', 'First dose', 'Country/ Territory', '2nd additional dose'}. Present columns are: Index(['date,location,total_vaccinations,people_vaccinated,people_fully_vaccinated,total_boosters,new_vaccinations,new_vaccinations_interpolated,new_people_vaccinated_interpolated,new_vaccinations_smoothed,new_people_vaccinated_smoothed', 'date'], dtype='object')
","Missing column fields: {'1st additional dose', 'Second dose', 'Total doses', 'Country code', 'Single dose', 'First dose', 'Country/ Territory', '2nd additional dose'}. Present columns are: Index(['date,location,total_vaccinations,people_vaccinated,people_fully_vaccinated,total_boosters,new_vaccinations,new_vaccinations_interpolated,new_people_vaccinated_interpolated,new_vaccinations_smoothed,new_people_vaccinated_smoothed', 'date'], dtype='object')"
cowidev.vax.incremental.pakistan,,,2023-07-13T07:00:22,,
cowidev.vax.incremental.philippines,,,2023-07-13T07:00:22,,
cowidev.vax.incremental.poland,2.66,True,2023-07-13T07:00:25,,
cowidev.vax.incremental.qatar,,,2023-07-13T07:00:22,,
cowidev.vax.incremental.russia,24.93,True,2023-07-13T07:00:47,,
cowidev.vax.incremental.saint_lucia,,,2023-07-13T07:00:22,,
cowidev.vax.incremental.serbia,16.18,True,2023-07-13T07:00:39,,
cowidev.vax.incremental.south_africa,,,2023-07-13T07:00:23,,
cowidev.vax.incremental.spain,23.53,True,2023-07-13T07:00:46,,
cowidev.vax.incremental.sri_lanka,,,2023-07-13T07:00:24,,
cowidev.vax.incremental.suriname,16.26,True,2023-07-13T07:00:40,,
cowidev.vax.incremental.taiwan,16.86,True,2023-07-13T07:00:41,,
cowidev.vax.incremental.thailand,5.46,True,2023-07-13T07:00:30,,
cowidev.vax.incremental.turkey,,,2023-07-13T07:00:25,,
cowidev.vax.incremental.united_arab_emirates,,,2023-07-13T07:00:25,,
cowidev.vax.incremental.vietnam,,,2023-07-13T07:00:25,,
cowidev.vax.incremental.who,33.95,True,2023-07-13T07:00:59,,
cowidev.vax.incremental.zambia,1.93,True,2023-07-13T07:00:28,,
